File: apitools/api-tools.go
package apitools

import (
	"fmt"
	"github.com/paigeweber13/mining-profit/config"
	"io/ioutil"
	"net/http"
)

// MiningPoolHub
func MPHGetbalance() {
	const MPHUrlEthBalance = "https://ethereum.miningpoolhub.com/index.php?page=api&action=getuserbalance&api_key=%s"

	conf := config.LoadConfig()
	resp, err := http.Get(
		fmt.Sprintf(MPHUrlEthBalance, conf.MiningPoolHub.ApiKey))
	if err != nil {
		fmt.Printf("ERROR: %v", err)
	}

	defer resp.Body.Close()
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Printf("ERROR: %v", err)
	}

	fmt.Printf("%s\n", body)
}

File: miningprofit/apitools.py
import miningprofit.config
import requests
import urllib.parse

def miningpoolhub_get_balance(apikey):
    url_base = 'https://ethereum.miningpoolhub.com/index.php?'
    query_vars = {'page': 'api', 'action': 'getuserbalance', 
        'api_key': apikey}
    query_string = urllib.parse.urlencode(query_vars)
    url = "{}{}".format(url_base, query_string)

    r = requests.get(url)
    return r.json()['getuserbalance']['data']

def miningpoolhub_get_hashrate(apikey):
    url_base = 'https://ethereum.miningpoolhub.com/index.php?'
    query_vars = {'page': 'api', 'action': 'getuserhashrate', 
        'api_key': apikey}
    query_string = urllib.parse.urlencode(query_vars)
    url = "{}{}".format(url_base, query_string)

    r = requests.get(url)
    return r.json()['getuserhashrate']['data']
